### DISCUSSION TITLE
Weather in shipcard

### DISCUSSION THREAD

**Discussion by Taxom** [jvde-github/AIS-catcher#176] [Ideas]
Hi

Is it possible to add the weather that it broadcasts to the ship’s card?
And it is possible to return the display of weather stations (weather forecasts with coordinates). Just display them on the map not as a vessel, but a weather icon.

Thanks! MaxPi

**Comment by jvde-github** [comment]
Where is this weather data coming from? And what is the interface? If I have such a device might add it.

Or do you mean the message 8 buoy information?

**Comment by Taxom** [comment]
Not only buoys but also ships broadcast weather. As I understand it, this happens through messages type 8. 
Here is a screenshot from Windy where it can be seen that the weather information was received from the ship with the callsign: CFN4229.
![изображение](https://github.com/jvde-github/AIS-catcher/assets/20313019/12794cf1-5d48-4122-b707-90bc104c794d)
https://www.marinetraffic.com/en/ais/details/ships/shipid:380951/mmsi:316006868/imo:9127198/vessel:ALGOSEA

**Comment by Taxom** [comment]
This is from meteo buoy 
![изображение](https://github.com/jvde-github/AIS-catcher/assets/20313019/2e1976c4-ec42-4201-a6fb-8eec89f085d6)

**Comment by Taxom** [comment]
As well as the messages that we discussed here: https://github.com/jvde-github/AIS-catcher/discussions/121

**Comment by jvde-github** [comment]
it could be this message: https://www.e-navigation.nl/content/weather-observation-report-ship

I looked at a few examples from the earlier discussion and did not find them very interesting information-wise to spend a lot of time on it as half the data was not populated (plus I don't receive them on my own station).

Perhaps this message 8 is a bit more interesting. Let me see what I can pick up.

This is from your earlier example. The JSON decoder can provide the details, most info is useful if you are navigating these waters:
```
{"class":"AIS","device":"AIS-catcher","scaled":true,"channel":"A","nmea":["!AIVDM,2,1,0,A,8@30oo@0GufWD1?4S2A9gwvlFR06EuOwgwl?wnSwe7wvlOwwsAwwnSGm,0*41","!AIVDM,2,2,0,A,wvh0,0*4F"],"type":8,"repeat":1,"mmsi":3160029,"dac":1,"fid":31,"lon":-79.400002,"lat":43.183334,"accuracy":false,"day":18,"hour":4,"minute":38,"visgreater":0,"visibility":12.700000,"waterlevel":30.009998,"cspeed2":25.500000,"cdir2":360,"cdepth2":31,"cspeed3":25.500000,"cdir3":360,"cdepth3":31,"waveheight":25.500000,"waveperiod":63,"wavedir":360,"swellheight":25.500000,"swellperiod":63,"swelldir":360,"seastate":13}
```

**Reply by Taxom** [reply]
Well, or you live not far, information about the wind and temperature is needed not only on board. Also, if you use AisCatcher as a source of data on board, it will be useful. I often use data from the nearest weather stations to see current weather.

**Comment by Taxom** [comment]
https://academy.iala-aism.org/asm/page/4/

**Reply by jvde-github** [reply]
I will have a look when back home what I can pick up
